### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by nimalank7**
Routes and authenticated added to model as well as views

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -4,13 +4,27 @@
...
+  enable :sessions
 
   Database_connection.connect
 
   get '/' do
     erb(:login, {:layout => true})
   end
 
+  get '/login' do
+    erb(:login)
```
Comment (waynodiablo) [inline app.rb]: .erb for login needs to take in the layout template. See /spaces

File: app.rb [hunk 2]
```diff
@@ -4,13 +4,27 @@
...
+  end
+
+  post '/login' do
+    p params
```
Comment (waynodiablo) [inline app.rb]: Suggest to remove debugging code

File: app.rb [hunk 3]
```diff
@@ -4,13 +4,27 @@
...
+    if User.authenticate(params[:email], params[:password]) == false
+      redirect('/login')
+    end
+    session[:user] = User.authenticate(params[:email], params[:password])
+    redirect('/spaces') #Redirects
+  end
+
   get '/register' do
     erb(:register)
```
Comment (waynodiablo) [inline app.rb]: Edit for layout template

File: app.rb [hunk 4]
```diff
@@ -20,7 +34,7 @@ class Airbnb < Sinatra::Base
     if User.email_already_exists?(params[:Email]) == true
       redirect('/register') # Sinatra flash will have to be added later
     end
-    User.create(params[:Email], params[:Username], params[:Password])
+    session[:user] = User.create(params[:Email], params[:Username], params[:Name], params[:Password])
     redirect('/spaces')
   end
```
Comment (waynodiablo) [inline app.rb]: Style - consistency in type case used, i.e use lowercase in above for email, name and password?

File: spec/user_spec.rb [hunk 5]
```diff
@@ -19,31 +17,46 @@
   end
 
   it "creates a user with id 1" do
```
Comment (waynodiablo) [inline spec/user_spec.rb]: Test name needs changing to reflect test?

**Comment by GDSNewt** [comment]
:shipit:

**Pull request merged**